src/cmd/interactive/handler.rs

```diff
@@ -0,0 +1,156 @@
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+
+use std::path::PathBuf;
+use anyhow::Error;
+use futures::TryStreamExt;
+use http::{Method, Request, Response, StatusCode};
+use hyper::server::conn::{Http};
+use hyper::service::Service;
+use hyper::Body;
+use tokio::sync::Mutex;
+use std::thread::JoinHandle;
+use tokio::runtime::Runtime;
+use tracing::instrument;
+use tokio::net::{UnixListener};
+#[cfg(unix)]
+use crate::todarpc::TodaRpc;
+use crate::injector::{InjectorConfig};
+
+#[derive(Debug)]
+pub struct TodaServer {
+    toda_rpc: Arc<Mutex<TodaRpc>>,
+    task: Option<JoinHandle<Result<(), Error>>>,
+}
+
+impl TodaServer {
+    pub fn new(toda_rpc: TodaRpc) -> Self {
+        Self {
+            toda_rpc: Arc::new(Mutex::new(toda_rpc)),
+            task: None,
+        }
+    }
+
+    pub fn serve_interactive(&mut self, interactive_path: PathBuf) {
+
+        let toda_rpc = self.toda_rpc.clone();
+        self.task = Some(std::thread::spawn( move || {
```

Comment:
Maybe we can replace it with just `tokio::task::spawn` instead construct another runtime, as we use the default in the `main` function.